### PR TITLE
Manage organizers team

### LINKED ISSUE #74: [organizer] manage organizers team
- [x] organizations page
  - [x] create a new organization (add owner attribute in organization collection)
  - [x] list my organizations

- [x] page of an organization
  - [x] update organization name
  - [x] invite a member (co-speaker like)
  - [x] leave organization
  - [x] list members
  - [x] remove a member if you are owner

- [x] my events page
  - [x] aggregate owner events + organizations events

- [x] associate event to one of my organization (in create / update)

- [x] firestore
 ```
    /organizations/{id}
        - id
        - name
        - owner
```

 ```
    /users/{id}
        - organizations: {
              [id]: true
          }
```

 ```
    /events/{id}
        - organization: <id>
```

### PR DESCRIPTION
**Pull request by frinyvonnick**
Fix #74

### CODE REVIEW

File: src/screens/organizer/organizations/organizations.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import Titlebar from 'components/titlebar'
+
+const MyOrganizations = () => (
+  <div className="events-page">
+    <Titlebar className="events-header" icon="fa fa-users" title="My organizations" />
```
Comment (bpetetot) [inline src/screens/organizer/organizations/organizations.jsx]: don't forget to use classNames dedicated for organizations (here it's event page classNames)

**Review by bpetetot — APPROVED** [review]
👌 nice starts ;)

**Review by bpetetot — APPROVED** [review]

File: src/store/listeners/listeners.js [hunk 2]
```diff
@@ -27,6 +28,7 @@ export default [
   /* user */
   when('@@ui/FETCH_USER')(user.fetchUser),
   when('@@ui/SAVE_PROFILE')(user.saveProfile),
+  when(/@@ui\/(.*)_ORGANIZATION_TO_USER/g)(user.updateOrganizationToUser),
```
Comment (fabienjuif) [inline src/store/listeners/listeners.js]: you don't need to catch the variable `()`, nor use the `/g` I think

`when(/@@ui\/.*_ORGANIZATION_TO_USER/)`

File: src/store/listeners/reactions/organizations.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  // FIXME: https://github.com/bpetetot/conference-hall/issues/167
+  const ref = await organizationCrud.read(id)
+  organizationCrud.update({ id: ref.id, ...ref.data() })
+
+  // Add organization id in owner organizations property
+  store.dispatch({ type: '@@ui/ADD_ORGANIZATION_TO_USER', payload: { uid, organizationId: ref.id } })
+
+  // FIXME: Go to newly created organization page
+  // go to organization page
+  store.dispatch(push('/organizer/organizations'))
```
Comment (fabienjuif) [inline src/store/listeners/reactions/organizations.js]: do you know about this ? https://github.com/alakarteio/k-ramel/tree/master/packages/drivers/redux-little-router

@bpetetot

File: src/store/listeners/reactions/user.js [hunk 4]
```diff
@@ -23,3 +24,23 @@ export const saveProfile = reaction((action, store, { form }) => {
...
+  const { uid, organizationId } = action.payload
+
+  let user = store.data.users.get(uid)
+  if (!user) {
+    const ref = await userCrud.read(uid)
+    user = ref.data()
+  }
+
+  let updated
+  if (action.type === '@@ui/ADD_ORGANIZATION_TO_USER') {
```
Comment (fabienjuif) [inline src/store/listeners/reactions/user.js]: I think this is better to have 2 reations

File: src/store/listeners/reactions/organizations.js (hunk 3 above)
Comment (fabienjuif) [reply src/store/listeners/reactions/organizations.js]: related to https://github.com/bpetetot/conference-hall/pull/163 (that I forgot sorry)

File: src/screens/organizer/organizations/organizations.container.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import forRoute from 'hoc-little-router'
+
+import loader from 'components/loader'
+import Organizations from './organizations'
+
+const mapStore = store => ({
+  loaded: store.ui.organizer.myOrganizations.isInitialized(),
+  organizations: store.ui.organizer.myOrganizations.getAsArray(),
+  load: () => store.dispatch('@@ui/ON_LOAD_ORGANIZER_ORGANIZATIONS'),
+  onSelect: organizationId => store.dispatch(push(`/organizer/organizations/${organizationId}`)),
```
Comment (fabienjuif) [inline src/screens/organizer/organizations/organizations.container.js]: there is redux-little-router driver

`mapStore = (store, ownProps, drivers) => ({`

File: src/screens/organizer/organizations/organizations.container.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+})
+
+export default compose(
+  forRoute.absolute('HOME_ORGANIZATION'), //
```
Comment (fabienjuif) [inline src/screens/organizer/organizations/organizations.container.js]: I have the feeling that `//` is for prettier.
Can't you configure it ? Is there one specific rule ?

(I really don't like auto-formatting solutions, I hated that with auto-format from eclipse, I dislike that with prettier)

File: src/screens/organizer/organizations/organizations.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    <Titlebar className="organizations-header" icon="fa fa-users" title="My organizations">
+      <Link href="/organizer/organization/create" className="btn">
+        <IconLabel icon="fa fa-users" label="Create organization" />
+      </Link>
+    </Titlebar>
+    <List
+      className="organizations-content"
+      array={organizations}
+      noResult="No organization yet !"
+      renderRow={({ id, name, updateTimestamp }) => (
```
Comment (fabienjuif) [inline src/screens/organizer/organizations/organizations.jsx:20]: Why don't you write a component ?

File: src/store/listeners/reactions/organizations.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  const { uid } = store.auth.get()
+
+  let user = store.data.users.get(uid)
+  if (!user) {
+    const ref = await userCrud.read(uid)
+    user = { id: ref.id, ...ref.data() }
+    store.data.users.add(user)
+  }
+
+  const organizerOrganizations = await Promise.all(map(
```
Comment (fabienjuif) [inline src/store/listeners/reactions/organizations.js:52]: What about
```js
const organizerOrganizations = await Promise.all(map(
  user.organizations,
  async (value, id) => {
    let organization = store.data.organizations.get(id)
    if (organization) return organization

    const ref = await organizationCrud.read(organizationId)
    return { ...ref.data(), id }
  },
))

store.data.organizations.set(organizerOrganizations)
```

then have a listener
```js
when('@@krf/SET>DATA>ORGANIZATIONS')(uiUpdate)
```

With
```js
export const uiUpdate = ({ payload }, store) => {
  store.ui.organizer.myOrganizations.reset()
  store.ui.organizer.myOrganizations.set(payload)
}
```

File: src/store/listeners/reactions/user.js [hunk 9]
```diff
@@ -23,3 +24,33 @@ export const saveProfile = reaction((action, store, { form }) => {
...
+}
+
+export const addOrganizationToUser = reaction(async (action, store) => {
+  const { uid, organizationId } = action.payload
+  const user = getUser(uid)(store)
+
+  const updated = set(user, `organizations.${organizationId}`, true)
+
+  store.data.users.update(updated)
+  await userCrud.update(updated)
```
Comment (fabienjuif) [inline src/store/listeners/reactions/user.js:46]: Why are you waiting for this ? When there is nothing next ?

File: src/store/listeners/reactions/user.js [hunk 10]
```diff
@@ -23,3 +24,33 @@ export const saveProfile = reaction((action, store, { form }) => {
...
+})
+
+export const removeOrganizationToUser = reaction(async (action, store) => {
+  const { uid, organizationId } = action.payload
+  const user = getUser(uid)(store)
+
+  const updated = unset(user, `organizations.${organizationId}`)
+
+  store.data.users.update(updated)
+  await userCrud.update(updated)
```
Comment (fabienjuif) [inline src/store/listeners/reactions/user.js:61]: Why are you waiting for this ? When there is nothing next ?

File: src/screens/organizer/organizations/organizations.jsx (hunk 7 above)
Comment (frinyvonnick) [reply src/screens/organizer/organizations/organizations.jsx:20]: I did it in #209 👌

File: src/store/listeners/reactions/user.js (hunk 9 above)
Comment (frinyvonnick) [reply src/store/listeners/reactions/user.js:46]: Deprecated comment

File: src/store/listeners/reactions/user.js (hunk 10 above)
Comment (frinyvonnick) [reply src/store/listeners/reactions/user.js:61]: Deprecated comment

**Pull request merged**